### ISSUE TITLE
[BUG] Error creating bean with name 'simpleAasRepositoryFactory' (aas-environment, 2.0.0-milestone-03)

### ISSUE THREAD

**Issue by logmanoriginal** [eclipse-basyx/basyx-java-server-sdk#376]
**Describe the bug**
`aas-environment` fails to start when I update my docker containers from `2.0.0-SNAPSHOT` to `2.0.0-milestone-03`:

```
aas-env-1        | Loaded property file: file:application.properties
aas-env-1        | Loaded property file: jar:nested:/application/basyxExecutable.jar/!BOOT-INF/classes/!/application.properties
aas-env-1        |   ____           _____
aas-env-1        |  |  _ \         / ____|
aas-env-1        |  | |_) |  __ _ | (___   _   _ __  __
aas-env-1        |  |  _ <  / _` | \___ \ | | | |\ \/ /
aas-env-1        |  | |_) || (_| | ____) || |_| | >  <
aas-env-1        |  |____/  \__,_||_____/  \__, |/_/\_\
aas-env-1        | ======================== __/ |======
aas-env-1        | AAS Environment         |___/
aas-env-1        | 2.0.0-PREVIEW
aas-env-1        |
aas-env-1        | 2024-08-02T08:35:19.631Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentComponent      : Starting AasEnvironmentComponent v2.0.0-milestone-03 using Java 17.0.12 with PID 1 (/application/basyxExecutable.jar started by nobody in /application)
aas-env-1        | 2024-08-02T08:35:19.636Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentComponent      : No active profile set, falling back to 1 default profile: "default"
aas-env-1        | 2024-08-02T08:35:21.323Z  INFO 1 --- [AAS Environment] [           main] .s.d.r.c.RepositoryConfigurationDelegate : Bootstrapping Spring Data MongoDB repositories in DEFAULT mode.
aas-env-1        | 2024-08-02T08:35:21.335Z  INFO 1 --- [AAS Environment] [           main] .s.d.r.c.RepositoryConfigurationDelegate : Finished Spring Data repository scanning in 8 ms. Found 0 MongoDB repository interfaces.
aas-env-1        | 2024-08-02T08:35:21.381Z  INFO 1 --- [AAS Environment] [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'errorChannel' has been explicitly defined. Therefore, a default PublishSubscribeChannel will be created.
aas-env-1        | 2024-08-02T08:35:21.390Z  INFO 1 --- [AAS Environment] [           main] faultConfiguringBeanFactoryPostProcessor : No bean named 'integrationHeaderChannelRegistry' has been explicitly defined. Therefore, a default DefaultHeaderChannelRegistry will be created.
aas-env-1        | 2024-08-02T08:35:21.879Z  INFO 1 --- [AAS Environment] [           main] o.s.b.w.embedded.tomcat.TomcatWebServer  : Tomcat initialized with port 8081 (http)
aas-env-1        | 2024-08-02T08:35:21.892Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardService   : Starting service [Tomcat]
aas-env-1        | 2024-08-02T08:35:21.892Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardEngine    : Starting Servlet engine: [Apache Tomcat/10.1.26]
aas-env-1        | 2024-08-02T08:35:21.925Z  INFO 1 --- [AAS Environment] [           main] o.a.c.c.C.[Tomcat].[localhost].[/]       : Initializing Spring embedded WebApplicationContext
aas-env-1        | 2024-08-02T08:35:21.926Z  INFO 1 --- [AAS Environment] [           main] w.s.c.ServletWebServerApplicationContext : Root WebApplicationContext: initialization completed in 2179 ms
aas-env-1        | 2024-08-02T08:35:22.115Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentFeaturePrinter : -------------------- Aas Environment Features: --------------------
aas-env-1        | 2024-08-02T08:35:22.115Z  INFO 1 --- [AAS Environment] [           main] o.e.d.b.a.c.AasEnvironmentFeaturePrinter : -----------------------------------------------------------------
aas-env-1        | 2024-08-02T08:35:22.122Z  WARN 1 --- [AAS Environment] [           main] ConfigServletWebServerApplicationContext : Exception encountered during context initialization - cancelling refresh attempt: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
aas-env-1        | Related cause: org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
aas-env-1        | 2024-08-02T08:35:22.126Z  INFO 1 --- [AAS Environment] [           main] o.apache.catalina.core.StandardService   : Stopping service [Tomcat]
aas-env-1        | 2024-08-02T08:35:22.142Z  INFO 1 --- [AAS Environment] [           main] .s.b.a.l.ConditionEvaluationReportLogger :
aas-env-1        |
aas-env-1        | Error starting ApplicationContext. To display the condition evaluation report re-run your application with 'debug' enabled.
aas-env-1        | 2024-08-02T08:35:22.157Z ERROR 1 --- [AAS Environment] [           main] o.s.boot.SpringApplication               : Application run failed
aas-env-1        |
aas-env-1        | org.springframework.beans.factory.UnsatisfiedDependencyException: Error creating bean with name 'simpleAasRepositoryFactory' defined in URL [jar:nested:/application/basyxExecutable.jar/!BOOT-INF/lib/basyx.aasrepository-backend-2.0.0-milestone-03.jar!/org/eclipse/digitaltwin/basyx/aasrepository/backend/SimpleAasRepositoryFactory.class]: Unsatisfied dependency expressed through constructor parameter 0: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
aas-env-1        |      at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:795) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.ConstructorResolver.autowireConstructor(ConstructorResolver.java:237) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.autowireConstructor(AbstractAutowireCapableBeanFactory.java:1375) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1212) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.DefaultListableBeanFactory.preInstantiateSingletons(DefaultListableBeanFactory.java:975) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.context.support.AbstractApplicationContext.finishBeanFactoryInitialization(AbstractApplicationContext.java:971) ~[spring-context-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:625) ~[spring-context-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.boot.web.servlet.context.ServletWebServerApplicationContext.refresh(ServletWebServerApplicationContext.java:146) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.springframework.boot.SpringApplication.refresh(SpringApplication.java:754) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.springframework.boot.SpringApplication.refreshContext(SpringApplication.java:456) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.springframework.boot.SpringApplication.run(SpringApplication.java:335) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.springframework.boot.SpringApplication.run(SpringApplication.java:1363) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.springframework.boot.SpringApplication.run(SpringApplication.java:1352) ~[spring-boot-3.3.2.jar!/:3.3.2]
aas-env-1        |      at org.eclipse.digitaltwin.basyx.aasenvironment.component.AasEnvironmentComponent.main(AasEnvironmentComponent.java:42) ~[!/:2.0.0-milestone-03]
aas-env-1        |      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
aas-env-1        |      at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77) ~[na:na]
aas-env-1        |      at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
aas-env-1        |      at java.base/java.lang.reflect.Method.invoke(Method.java:569) ~[na:na]
aas-env-1        |      at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:91) ~[basyxExecutable.jar:2.0.0-milestone-03]
aas-env-1        |      at org.springframework.boot.loader.launch.Launcher.launch(Launcher.java:53) ~[basyxExecutable.jar:2.0.0-milestone-03]
aas-env-1        |      at org.springframework.boot.loader.launch.JarLauncher.main(JarLauncher.java:58) ~[basyxExecutable.jar:2.0.0-milestone-03]
aas-env-1        | Caused by: org.springframework.beans.factory.BeanCreationException: Error creating bean with name 'aasMongoDBBackendProvider': Resolution of declared constructors on bean Class [org.eclipse.digitaltwin.basyx.aasrepository.backend.mongodb.AasMongoDBBackendProvider] from ClassLoader [org.springframework.boot.loader.launch.LaunchedClassLoader@2ff4acd0] failed
aas-env-1        |      at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:383) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.determineConstructorsFromBeanPostProcessors(AbstractAutowireCapableBeanFactory.java:1314) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBeanInstance(AbstractAutowireCapableBeanFactory.java:1209) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.doCreateBean(AbstractAutowireCapableBeanFactory.java:562) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractAutowireCapableBeanFactory.createBean(AbstractAutowireCapableBeanFactory.java:522) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.lambda$doGetBean$0(AbstractBeanFactory.java:337) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.DefaultSingletonBeanRegistry.getSingleton(DefaultSingletonBeanRegistry.java:234) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.doGetBean(AbstractBeanFactory.java:335) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.AbstractBeanFactory.getBean(AbstractBeanFactory.java:200) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.config.DependencyDescriptor.resolveCandidate(DependencyDescriptor.java:254) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.DefaultListableBeanFactory.doResolveDependency(DefaultListableBeanFactory.java:1443) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.DefaultListableBeanFactory.resolveDependency(DefaultListableBeanFactory.java:1353) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.ConstructorResolver.resolveAutowiredArgument(ConstructorResolver.java:904) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      at org.springframework.beans.factory.support.ConstructorResolver.createArgumentArray(ConstructorResolver.java:782) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      ... 26 common frames omitted
aas-env-1        | Caused by: java.lang.NoClassDefFoundError: org/eclipse/digitaltwin/basyx/common/mongocore/BasyxMongoMappingContext
aas-env-1        |      at java.base/java.lang.Class.getDeclaredConstructors0(Native Method) ~[na:na]
aas-env-1        |      at java.base/java.lang.Class.privateGetDeclaredConstructors(Class.java:3373) ~[na:na]
aas-env-1        |      at java.base/java.lang.Class.getDeclaredConstructors(Class.java:2555) ~[na:na]
aas-env-1        |      at org.springframework.beans.factory.annotation.AutowiredAnnotationBeanPostProcessor.determineCandidateConstructors(AutowiredAnnotationBeanPostProcessor.java:378) ~[spring-beans-6.1.11.jar!/:6.1.11]
aas-env-1        |      ... 39 common frames omitted
aas-env-1        | Caused by: java.lang.ClassNotFoundException: org.eclipse.digitaltwin.basyx.common.mongocore.BasyxMongoMappingContext
aas-env-1        |      at java.base/java.net.URLClassLoader.findClass(URLClassLoader.java:445) ~[na:na]
aas-env-1        |      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:592) ~[na:na]
aas-env-1        |      at org.springframework.boot.loader.net.protocol.jar.JarUrlClassLoader.loadClass(JarUrlClassLoader.java:103) ~[basyxExecutable.jar:2.0.0-milestone-03]
aas-env-1        |      at org.springframework.boot.loader.launch.LaunchedClassLoader.loadClass(LaunchedClassLoader.java:91) ~[basyxExecutable.jar:2.0.0-milestone-03]
aas-env-1        |      at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:525) ~[na:na]
aas-env-1        |      ... 43 common frames omitted
aas-env-1        |
Gracefully stopping... (press Ctrl+C again to force)
dependency failed to start: container basyxminimal-aas-env-1 exited (1)
```

**To Reproduce**
1. Copy the container example: [BaSyxMinimal](https://github.com/eclipse-basyx/basyx-java-server-sdk/tree/3446ff3b03b30e6398b996bbed7c45671a131c99/examples/BaSyxMinimal)
2. Replace container images `2.0.0-SNAPSHOT` with `2.0.0-milestone-03`
3. Run `docker compose up`

**Expected behavior**
All containers start successfully.

**BaSyx (please complete the following information):**
- BaSyx Version: 2.0.0-milestone-03
- BaSyx SDK: ---
- AAS Version: V3

**BaSyx configuration**
See [BaSyxMinimal](https://github.com/eclipse-basyx/basyx-java-server-sdk/tree/3446ff3b03b30e6398b996bbed7c45671a131c99/examples/BaSyxMinimal)

**System (please complete the following information):**
 - OS: Ubuntu-24.04 LTS (running in WSL2 on Windows 11)
 - CPU architecture: x64
 - Browser: ---
 - Version: ---

**Additional context**

**Comment by mdanish98** [comment]
Hi @logmanoriginal 

Thanks for raising this issue.
This issue is similar to #378, we will update the analysis as soon as possible.

**Comment by FriedJannik** [comment]
Hi @logmanoriginal,
Thank you for the Report.
This was indeed a bug which was fixed in the PR mentioned above #383 
We are planning to release a new image with the fix included.

**Comment by logmanoriginal** [comment]
This is fixed in `2.0.0-milestone-03.1`. Thanks for the quick response!
